src/Microsoft.AspNetCore.Mvc.ApiExplorer/DefaultApiDescriptionProvider.cs

```diff
@@ -382,6 +375,15 @@ private string GetRelativePath(RouteTemplate parsedTemplate)
                 }
             }
 
+            // Set the default status only when no status has already been set explicitly
+            if (objectTypes.Count == 0)
+            {
+                if (type != null && type != typeof(void))
+                {
+                    objectTypes[StatusCodes.Status200OK] = type;
+                }
+            }
```

Comment:
Can you open a bug for this? I don't want to bloat this PR, but it seems wrong that we infer 'default 200' when you a DTO as your return type and 'nothing' when you have void as your return type.